test/test_sparse.py

```diff
@@ -227,8 +228,11 @@ def test_tensor(x, res):
             x.to_dense()  # Tests triple to_dense for memory corruption
             x.to_dense()
             x.to_dense()
-            self.assertEqual(res, x.to_dense())
-            self.assertEqual(res, self.safeToDense(x))
+            # We dont have to_dense for half types, so we don't request 
```

Comment:
if the problem is only with half, why not use `exact_dtype=x.dtype is not torch.float16` ?